File: src/main/java/com/paragon/client/systems/module/hud/impl/Crystals.java
package com.paragon.client.systems.module.hud.impl;

import com.paragon.api.util.render.RenderUtil;
import com.paragon.client.systems.module.hud.HUDModule;
import com.paragon.client.systems.module.impl.client.Colours;
import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraft.util.math.MathHelper;

public class Crystals extends HUDModule {

    public static Crystals INSTANCE;

    public Crystals() {
        super("Crystals", "Displays the amount of crystals in your inventory");

        INSTANCE = this;
    }

    @Override
    public void render() {
        RenderUtil.drawRect(getX(), getY(), getWidth(), getHeight(), 0x70000000);
        RenderUtil.drawBorder(getX(), getY(), getWidth(), getHeight(), 1, Colours.mainColour.getValue().getRGB());

        ItemStack itemStack = new ItemStack(Items.END_CRYSTAL, getCrystals());

        RenderUtil.renderItemStack(itemStack, getX() + getWidth() - 18, getY() + 2, true);
    }

    @Override
    public float getWidth() {
        return MathHelper.clamp(mc.fontRenderer.getStringWidth(String.valueOf(getCrystals())), 18, mc.fontRenderer.getStringWidth(String.valueOf(getCrystals()))) + 2;
    }

    @Override
    public float getHeight() {
        return 19;
    }

    public int getCrystals() {
        int count = 0;

        for (int i = 0; i < 36; i++) {
            ItemStack itemStack = mc.player.inventory.getStackInSlot(i);

            if (itemStack.getItem() == Items.END_CRYSTAL) {
                count += itemStack.getCount();
            }
        }

        if (mc.player.getHeldItemMainhand().getItem() == Items.END_CRYSTAL) {
            count += mc.player.getHeldItemMainhand().getCount();
        }

        if (mc.player.getHeldItemOffhand().getItem() == Items.END_CRYSTAL) {
            count += mc.player.getHeldItemOffhand().getCount();
        }

        return count;
    }
}

File: src/main/kotlin/com/paragon/client/systems/module/hud/impl/Crystals.kt
package com.paragon.client.systems.module.hud.impl

import com.paragon.api.util.render.RenderUtil.drawBorder
import com.paragon.api.util.render.RenderUtil.drawRect
import com.paragon.api.util.render.RenderUtil.renderItemStack
import com.paragon.client.systems.module.hud.HUDModule
import com.paragon.client.systems.module.impl.client.Colours
import net.minecraft.init.Items
import net.minecraft.item.ItemStack
import net.minecraft.util.math.MathHelper

object Crystals : HUDModule("Crystals", "Displays the amount of crystals in your inventory") {

    override fun render() {
        drawRect(x, y, width, height, 0x70000000)
        drawBorder(x, y, width, height, 1f, Colours.mainColour.value.rgb)
        val itemStack = ItemStack(Items.END_CRYSTAL, crystals)
        renderItemStack(itemStack, x + width - 18, y + 2, true)
    }

    override var width: Float
        get() = (MathHelper.clamp(minecraft.fontRenderer.getStringWidth(crystals.toString()), 19, minecraft.fontRenderer.getStringWidth(crystals.toString())) + 2).toFloat()

        set(width) {
            super.width = width
        }

    override var height: Float
        get() = 19f

        set(height) {
            super.height = height
        }

    val crystals: Int
        get() {
            var count = 0

            for (i in 0..35) {
                val itemStack: ItemStack = minecraft.player.inventory.getStackInSlot(i)

                if (itemStack.item === Items.END_CRYSTAL) {
                    count += itemStack.count
                }
            }

            if (minecraft.player.heldItemMainhand.item === Items.END_CRYSTAL) {
                count += minecraft.player.heldItemMainhand.count
            }

            if (minecraft.player.heldItemOffhand.item === Items.END_CRYSTAL) {
                count += minecraft.player.heldItemOffhand.count
            }

            return count
        }

}